### ISSUE TITLE
Change event yes/no items

### ISSUE THREAD

**Issue by davidskalinder** [davidskalinder/mpeds-coder#7]
About each event:
- Protest?
  - A protest (code event)
  - Not a protest (don't code)
  - Not sure (code event)
- Involve Black people, organizations, or issues? 
  - Black or anti-Black issue or protesters (including Black immigrants) (code)
  - No Black issue or protesters (don't code)
  - not sure (code)
- Is this event out of scope? 
  - Historical (occurred before 1980) (don't code)
  - Outside the US  (don't code). 
  - Not sure 
- Characteristics of this event (check all that apply)
  - The "event" is a reference to multiple events e.g. "protests last month" or "protests in 10 cities"
  - The "event" is complex and has sub-events that you did NOT code separately.
  - This event is part of a complex event or series of events that you decided to code as separate events.
  - A counter-protest (a response to some other group's protest)

**Comment by davidskalinder** [comment]
I'd suggest that anything that might result in a "don't code" instruction gets moved to earlier in the coding workflow...

**Comment by davidskalinder** [comment]
PEO's note says: "Also helpful to move this tab 2 info to tab 1 if possible. Each of these could be yes/no [so possible to click more than one in a set, even though the first two are logically"...

My questions:

1. Does this mean we want to move everything from the Yes/No tab into the first (now, "Basic Info") tab? If so, the tab might be too long for some windows, and at the moment (for me at least) it can't scroll...
2. Does this mean that we want both yes and no options for both, e.g., "a protest" and "not a protest"? so the answer to both of those questions could be "yes"?

**Comment by davidskalinder** [comment]
Okay, and now I can mention @olderwoman to get her response to the questions above (in https://github.com/davidskalinder/mpeds-coder/issues/7#issuecomment-556413858)...

**Comment by olderwoman** [comment]
At this point I've lost track of what this would look like so hard to answer. I think the front page needs the screen out questions. Not marking events because (check all that apply):  all events outside US; all events prior to 1980; no protest event; no Black participants or issues.

**Comment by davidskalinder** [comment]
@olderwoman, @matthewsmorgan, this is the other issue that I think we'll need to nail down soon.  We should bear in mind for this one that any information that's about the article rather than a particular event should go in the article-level info section that I haven't yet finished building for issue #4; so stuff like "here's why I haven't coded any events" should presumably be entered at the article level and therefore not included in the list above.

**Comment by davidskalinder** [comment]
From meeting discussion:

About each event:
- Characteristics of this event (check all that apply)
  - The "event" is a reference to multiple events e.g. "protests last month" or "protests in 10 cities"
  - The "event" is complex and has sub-events that you did NOT code separately.
  - This event is part of a complex event or series of events that you decided to code as separate events.
  - A counter-protest (a response to some other group's protest)

**Comment by davidskalinder** [comment]
List is good enough for now -- put in testing instance, then bug @olderwoman (and @matthewsmorgan) daily until they've tested it.

**Comment by davidskalinder** [comment]
Changes to static items now done in branch remove_yesno_radios and merged into testing branch.  YAML has been changed in testing deployment.  This should be done pending testing (and, I suspect, some text revision)

**Comment by davidskalinder** [comment]
Okay @olderwoman and @matthewsmorgan, this should now be up-to-date and available on the testing server (NB: not the development server we used this afternoon! -- logins for the testing server can be found in the doc at Technical documentation > MAI).

Note that:

- The stuff that will eventually be collected at the article level (not a protest, etc.) isn't there yet
- What is there looks okay, but awfully wordy.  I left in all the explanatory clauses, but maybe it's better to take them out?  I defer to y'all's judgement.

And I guess now as requested I should set a daily reminder for myself to nag you two to test this until you're happy with it? :)

**Comment by olderwoman** [comment]
Hi, David, message received. I’ll try to take a look tonight. Also, you should try coding a dozen or so articles yourself, if you have not done so, as that will give you your own ideas about how to make this work easier.
Pam

**Comment by matthewsmorgan** [comment]
Writing my notes on trying out the Testing version today (some things are not related to this issue item, but I trust you will sort that out?):

1) Basic Info
- The date instructions still need to be added, and it looks like there's space for this. I suggest: "Start Date (year-month-day)
- Question: What is the purpose of "Other Location" under "City/Town"? Maybe this was put here before we remembered that Location should be on the text select list? If that's the case, I think this field is redundant and can be removed given that info such as city hall can be added under the text select section

2) Yes/No
- I'm fine with the instructions being there, but there needs to be an easy/quick way to identify the item. I suggest adding the following short descriptions (maybe in all caps) before the text that is currently there, and put that text in parentheses:
Multiple Events
Complex Event (uncoded)
Complex Event (coded)
Counter Protest
Campaign
Future Event

3) Text Select
I'd suggest shortening three of these items:
Actions by Protesters
Size/Number of Participants
Target

Also, text select was not working at all.

**Comment by davidskalinder** [comment]
Thanks @matthewsmorgan!  Responses below:

>     * The date instructions still need to be added, and it looks like there's space for this. I suggest: "Start Date (year-month-day)

Yep, this is waiting on #18, which I haven't gotten around to yet.

>     * Question: What is the purpose of "Other Location" under "City/Town"? Maybe this was put here before we remembered that Location should be on the text select list? If that's the case, I think this field is redundant and can be removed given that info such as city hall can be added under the text select section

This was something that was suggested in the discussion at #14 (to handle locations like "City Hall").  I'll move this comment there and reply to it there so that we can keep track of it...

>     1. Yes/No
>     * I'm fine with the instructions being there, but there needs to be an easy/quick way to identify the item. I suggest adding the following short descriptions (maybe in all caps) before the text that is currently there, and put that text in parentheses:
>       Multiple Events
>       Complex Event (uncoded)
>       Complex Event (coded)
>       Counter Protest
>       Campaign
>       Future Event

Sounds like it might be good enough for now.  The one thing I'd suggest is to use "Complex Event Parent" instead of Complex Event (uncoded) and "Complex Event Child" instead of Compex Event (coded) -- I think that captures the meanings of those items more precisely?

@olderwoman, do you have a different opinion, or should I go ahead and change this to the above?

>     1. Text Select
>        I'd suggest shortening three of these items:
>        Actions by Protesters
>        Size/Number of Participants
>        Target

I'll move this to #8.

> Also, text select was not working at all.

That is a surprise!  It works for me...?  Are you using Firefox?  (I don't know for sure, but this might be one of the things that doesn't work in other browsers?)  If not then we should open another issue for this and get to the bottom of it ASAP...

**Comment by matthewsmorgan** [comment]
@davidskalinder -- you're right, I totally forgot to open it in Firefox. Everything else worked so well in Chrome, hah! Sorry to scare you.

**Comment by davidskalinder** [comment]
Hah, excellent!

**Comment by olderwoman** [comment]
The only thing I saw about yes/no was that I often had nothing to check. Wondering whether to add "none of these" just so we know the coder went through the list.

**Comment by davidskalinder** [comment]
Might not be a bad idea, if we want to encourage them to look through every question every time?  @matthewsmorgan, what say you?

**Comment by matthewsmorgan** [comment]
Yes, I would be for adding a "None of the above"

**Comment by davidskalinder** [comment]
Okay, this should be up to date now in the testing deployment.  Let me know how it looks.

I see that for one thing the item panel is getting long enough that the scrolling issue is becoming more important...

**Comment by matthewsmorgan** [comment]
This works for me. Only possible suggestion for improvement is making "None of These" and other boxes mutually exclusive. Right now it looks like you can check all boxes. But I don't think this is super necessary.

**Comment by matthewsmorgan** [comment]
To clarify, I mean making None exclusive from all others. You should be able to check multiple boxes here, but it doesn't make sense to be able to mark None AND Multiple Events. However, like I said, in practice this may not matter much.

**Comment by davidskalinder** [comment]
> To clarify, I mean making None exclusive from all others. You should be able to check multiple boxes here, but it doesn't make sense to be able to mark None AND Multiple Events. However, like I said, in practice this may not matter much.

Yeah, I see what you mean.  Unfortunately, this would be tricky to implement since we'd have to split it into two groups: one for "None of these" and one for everything else, in which only one group or the other can have any boxes checked.  There might be some of Alex's code that does this, but it'd require getting hands a lot dirtier than just changing the text.

So given the time, complexity, and fairly low value (since we mostly intended the "None of these" box merely as a proof that coders didn't skip the section), I'm inclined to leave it alone, for now at least?  But if the value is high let me know and I can roll up my sleeves...

**Comment by olderwoman** [comment]
I’m ok with this resolution

**Comment by davidskalinder** [comment]
Notes from today's discussion on this:

- PEO/MM need to review yes-nos more seriously (so they just did it)
  - Complex events?
    - Coders are bad at this
  - Multiple events
    - Not sure whether p1 or p2 coders should do this
    - But these are not all that bad
  - Counter-event might need to be changed
    - Most helpful to know if things are anti-black or pro-black
    - Trick is that some protests are not anti-black, but pro-black protesters are anti-it (protests to which pro-black actors are opposed)
    - E.g., conservative events (Tea Party?)
    - OTOH, students seem to be pretty good at marking counter-events
  - What’s the point of these fields anyway?
    - Flagging to speed up p2
    - Calls p1 attention to what kind of protests to code
  - So: drop complex event items; keep counter-protest, multiple events; make campaign “part of campaign or episode”; add event counter to black interest

**Comment by davidskalinder** [comment]
Okay @olderwoman and @matthewsmorgan, these changes should now be live in the testing version; most were easy except for the wording of the "counter to black interest" item, which I've worded as "OPPOSED TO BLACK INTERESTS (This event's participants oppose the interests of black protesters)".  Let me know what you think.

**Comment by olderwoman** [comment]
I like your wording;

**Comment by davidskalinder** [comment]
NB this didn't quite make it into production on time: this was resolved late by #50.
